Core/include/Input.h

```diff
@@ -1,87 +1,82 @@
 #pragma once
-#include <unordered_map>
-#include <glad/glad.h> // Include glad first
+#include <glad/glad.h>
 #include <GLFW/glfw3.h>
+#include <queue>
 #include <iostream>
 
+#include "include/Body.h"
 
 class Input
 {
 public:
-    // static maps that keep track of the current state of keypresses & mouse clicks (pressed or unpress)
-    static std::unordered_map<int, bool> keys;
-    static std::unordered_map<int, bool> mouseButtons;
+    enum ActionType { PRESS, RELEASE, HOLD };
 
+    struct InputEvent {
+        int key;
+        ActionType action;
+    };
 
-    // Key callback function for GLFW
+    // Queue to store input events
+    static std::queue<InputEvent> eventQueue;
+
+    // Key callback functions for GLFW
     static void KeyCallback(GLFWwindow* window, int key, int scancode, int action, int mods)
     {
-        if (action == GLFW_PRESS)
-        {
-            keys[key] = true;
-        }
-        else if (action == GLFW_RELEASE)
-        {
-            keys[key] = false;
-        }
+        ActionType actionType = (action == GLFW_PRESS) ? PRESS : (action == GLFW_RELEASE ? RELEASE : HOLD);
+        eventQueue.push({ key, actionType });
     }
 
-    // Mouse callback function for GLFW
+    // Mouse callback for GLFW 
     static void MouseButtonCallback(GLFWwindow* window, int button, int action, int mods)
     {
-        if (action == GLFW_PRESS)
-        {
-            mouseButtons[button] = true;
-        }
-        else if (action == GLFW_RELEASE)
-        {
-            mouseButtons[button] = false;
-        }
-    }
-
-    static bool IsKeyPressed(int key)
-    {
-        return keys[key];
+        ActionType actionType = (action == GLFW_PRESS) ? PRESS : (action == GLFW_RELEASE ? RELEASE : HOLD);
+        eventQueue.push({ button, actionType });
     }
-
-    static bool IsMouseButtonPressed(int button)
+    
+    static void RunInputListener(AE86::RigidBody* rigidBody)
```

Comment:
How will we handle additional components that use input?
Should be converted to a switch-case for faster parsing